crates/assert-instr-macro/src/lib.rs

```diff
@@ -100,7 +100,8 @@ pub fn assert_instr(
     let to_test = quote! {
         #attrs
         #[no_mangle]
-        unsafe extern #abi fn #shim_name(#(#inputs),*) #ret {
+        #[inline(never)]
+        pub unsafe extern #abi fn #shim_name(#(#inputs),*) #ret {
```

Comment:
I'm not really sure why this wasn't `#[inline(never)]` before, but we really do want to prevent this shim from being removed / inlined.